src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFilterTelemetry.cs

```diff
@@ -0,0 +1,115 @@
+using System.Text;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement
+{
+    /// <summary>
+    /// Telemetry for tracking built-in feature filter usage.
+    /// </summary>
+    internal class FeatureFilterTelemetry
+    {
+        private const string CustomFilter = "CSTM";
+        private const string PercentageFilter = "PRCNT";
+        private const string TimeWindowFilter = "TIME";
+        private const string TargetingFilter = "TRGT";
+        private const string FilterTypeDelimiter = "+";
+
+        // Built-in Feature Filter Names
+        private readonly List<string> PercentageFilterNames = new List<string> { "Percentage", "Microsoft.Percentage", "PercentageFilter", "Microsoft.PercentageFilter" };
+        private readonly List<string> TimeWindowFilterNames = new List<string> { "TimeWindow", "Microsoft.TimeWindow", "TimeWindowFilter", "Microsoft.TimeWindowFilter" };
+        private readonly List<string> TargetingFilterNames = new List<string> { "Targeting", "Microsoft.Targeting", "TargetingFilter", "Microsoft.TargetingFilter" };
+
+        public bool UsesCustomFilter { get; set; } = false;
+        public bool UsesPercentageFilter { get; set; } = false;
+        public bool UsesTimeWindowFilter { get; set; } = false;
+        public bool UsesTargetingFilter { get; set; } = false;
+       
+        public bool UsesAnyFeatureFilter()
+        {
+            return UsesCustomFilter || UsesPercentageFilter || UsesTimeWindowFilter || UsesTargetingFilter;
+        }
+
+        public void ResetFeatureFilterTelemetry()
+        {
+            UsesCustomFilter = false;
+            UsesPercentageFilter = false;
+            UsesTimeWindowFilter = false;
+            UsesTargetingFilter = false;
+        }
+
+        public void UpdateFeatureFilterTelemetry(string filterName)
+        {
+            if (PercentageFilterNames.Any(name => string.Equals(name, filterName, StringComparison.OrdinalIgnoreCase)))
+            {
+                UsesPercentageFilter = true;
+            }
+            else if (TimeWindowFilterNames.Any(name => string.Equals(name, filterName, StringComparison.OrdinalIgnoreCase)))
+            {
+                UsesTimeWindowFilter = true;
+            }
+            else if (TargetingFilterNames.Any(name => string.Equals(name, filterName, StringComparison.OrdinalIgnoreCase)))
+            {
+                UsesTargetingFilter = true;
+            }
+            else
+            {
+                UsesCustomFilter = true;
+            }
+        }
+
+        /// <summary>
+        /// Returns a formatted string containing code names, indicating which feature filters are used by the application.
+        /// </summary>
+        /// <returns>Formatted string like: "CSTM+PRCNT+TIME+TRGT", "PRCNT+TRGT", etc. If no filters are used, empty string will be returned.</returns>
+        public override string ToString()
+        {
+            if (!UsesAnyFeatureFilter())
```

Comment:
Can you add the brackets for the `if`, it's really easy for weird things to happen when updating code that has bracketless ifs.